opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -407,26 +409,39 @@ private static JSONObject getCheckBoxJsonObjects(Event event, JSONObject parentO
 
     public static void addObservation(Event e, JSONObject jsonObject) {
         String value = getString(jsonObject, VALUE);
-        if (StringUtils.isBlank(value)) { return; }
+        if (StringUtils.isBlank(value)) {
+            return;
+        }
 
         String type = getString(jsonObject, AllConstants.TYPE);
         if (AllConstants.CHECK_BOX.equals(type)) {
             try {
                 List<Object> optionValues = new ArrayList<>();
+                List<Object> optionEntityIds = new ArrayList<>();
                 Map<String, Object> optionKeyVals = new HashMap<>();
                 if (jsonObject.has(AllConstants.OPTIONS)) {
                     JSONArray options = jsonObject.getJSONArray(AllConstants.OPTIONS);
+                    boolean shouldBeCombined = jsonObject.optBoolean(AllConstants.COMBINE_CHECKBOX_OPTION_VALUES);
+                    String entity = getString(jsonObject, OPENMRS_ENTITY);
                     for (int i = 0; i < options.length(); i++) {
                         JSONObject option = options.getJSONObject(i);
                         boolean optionValue = option.optBoolean(VALUE);
                         if (!optionValue) {
                             continue;
                         }
-                        if (CONCEPT.equals(getString(jsonObject, OPENMRS_ENTITY))) {
+                        if (CONCEPT.equals(entity)) {
+                            String optionKey = option.optString(KEY);
+                            String openmrsEntityId = option.optString(OPENMRS_ENTITY_ID);
```

Comment:
Should these be part of the if block? Is it used elsewhere?

Reply:
yeah it should be part since, it checks first if it is a concept before starting the combination

Reply:
Sorry, meant should it move to the if block at line 440?

Reply:
ok changing